src/pages/index.js

```diff
@@ -25,26 +27,64 @@ import {
   cardSelector,
 } from "../utils/constants.js";
 
+const avatarEditButton = document.querySelector("#avatar-edit-button");
+const avatarForm = document.querySelector("#modal__form-avatar");
+const profileImage = document.querySelector(".profile__image");
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "65820eda-9ddf-4d15-8d65-d12582853dbd",
+    "Content-Type": "application/json",
+  },
+});
+
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
 const handleProfileAddSubmit = ({ name, link }) => {
-  addCardForm.reset();
-  renderCard({ name, link });
-  // closePopup(profileAddModal);
-  formValidators["add-card-form"].disabledSubmitButton();
+  addCardPopup.renderLoading(true);
+  api
+    .addNewCard({ name, link })
+    .then((cardData) => {
+      renderCard(cardData);
+      addCardPopup.close();
+      formValidators["add-card-form"].disabledSubmitButton();
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`Error ${err}. Could not add card.`);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
 };
 
 const addCardPopup = new PopupWithForm(
   "#add-card-modal",
   handleProfileAddSubmit
 );
-addCardPopup.setEventListeners();
+addCardPopup.setEventListeners(); //indented
 
 const handleProfileEditSubmit = ({ title, description }) => {
-  userInfo.setUserInfo({ name: title, description: description });
+  profileEditPopup.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```